kurtosis-devnet/pkg/kurtosis/sources/spec/spec.go

```diff
@@ -0,0 +1,81 @@
+package spec
+
+import (
+	"fmt"
+	"io"
+
+	"gopkg.in/yaml.v3"
+)
+
+// ChainSpec represents the network parameters for a chain
+type ChainSpec struct {
+	Name      string
+	NetworkID string
+}
+
+// EnclaveSpec represents the parsed chain specifications from the YAML
+type EnclaveSpec struct {
+	Chains []ChainSpec
+}
+
+// NetworkParams represents the network parameters section in the YAML
+type NetworkParams struct {
+	Name      string `yaml:"name"`
+	NetworkID string `yaml:"network_id"`
+}
+
+// ChainConfig represents a chain configuration in the YAML
+type ChainConfig struct {
+	NetworkParams NetworkParams `yaml:"network_params"`
+}
+
+// OptimismPackage represents the optimism_package section in the YAML
+type OptimismPackage struct {
+	Chains []ChainConfig `yaml:"chains"`
+}
+
+// YAMLSpec represents the root of the YAML document
+type YAMLSpec struct {
+	OptimismPackage OptimismPackage `yaml:"optimism_package"`
+}
+
+type Spec struct{}
+
+type SpecOption func(*Spec)
+
+func NewSpec(opts ...SpecOption) *Spec {
+	s := &Spec{}
+	for _, opt := range opts {
+		opt(s)
+	}
+	return s
+}
+
+// ExtractData parses a YAML document and returns the chain specifications
+func (s *Spec) ExtractData(r io.Reader) (*EnclaveSpec, error) {
+	var yamlSpec YAMLSpec
+	decoder := yaml.NewDecoder(r)
+	if err := decoder.Decode(&yamlSpec); err != nil {
+		if err == io.EOF {
+			// Return empty spec for empty input
+			return &EnclaveSpec{
```

Comment:
Why do we do this rather than returning an error?

Reply:
My mistake, I was assuming that an empty document would lead to the creation of 0 L2 chains, but that's not even true given the existing defaults. I'll return an error indeed